infrastructure/lib/atat-web-api-stack.ts

```diff
@@ -34,125 +34,91 @@ export class AtatWebApiStack extends cdk.Stack {
-      resource: portfolioDrafts,
+    const createPortfolioDraft = new ApiDynamoDBFunction(this, "CreatePortfolioDraftFunction", {
       table: table,
       method: HttpMethod.POST,
       handlerPath: packageRoot() + "/api/portfolioDrafts/createPortfolioDraft.ts",
     });
 
-    const getPortfolioDrafts = new ApiDynamoDBFunction(this, "GetPortfolioDrafts", {
-      resource: portfolioDrafts,
+    const getPortfolioDrafts = new ApiDynamoDBFunction(this, "GetPortfolioDraftsFunction", {
       table: table,
       method: HttpMethod.GET,
       handlerPath: packageRoot() + "/api/portfolioDrafts/getPortfolioDrafts.ts",
     });
 
-    const deletePortfolioDraft = new ApiDynamoDBFunction(this, "DeletePortfolioDraft", {
-      resource: portfolioDraftId,
+    const deletePortfolioDraft = new ApiDynamoDBFunction(this, "DeletePortfolioDraftFunction", {
       table: table,
       method: HttpMethod.DELETE,
       handlerPath: packageRoot() + "/api/portfolioDrafts/deletePortfolioDraft.ts",
     });
 
-    const createPortfolioStep = new ApiDynamoDBFunction(this, "CreatePortfolioStep", {
-      resource: portfolio,
-      table: table,
-      method: HttpMethod.POST,
-      handlerPath: packageRoot() + "/api/portfolioDrafts/portfolio/createPortfolioStep.ts",
-    });
-
-    const getPortfolioStep = new ApiDynamoDBFunction(this, "GetPortfolioStep", {
-      resource: portfolio,
+    const getPortfolioStep = new ApiDynamoDBFunction(this, "GetPortfolioStepFunction", {
       table: table,
       method: HttpMethod.GET,
       handlerPath: packageRoot() + "/api/portfolioDrafts/portfolio/getPortfolioStep.ts",
     });
 
-    const createFundingStep = new ApiDynamoDBFunction(this, "CreateFundingStep", {
-      resource: funding,
+    const createFundingStep = new ApiDynamoDBFunction(this, "CreateFundingStepFunction", {
       table: table,
       method: HttpMethod.POST,
       handlerPath: packageRoot() + "/api/portfolioDrafts/funding/createFundingStep.ts",
     });
 
-    const getFundingStep = new ApiDynamoDBFunction(this, "GetFundingStep", {
-      resource: funding,
+    const getFundingStep = new ApiDynamoDBFunction(this, "GetFundingStepFunction", {
       table: table,
       method: HttpMethod.GET,
       handlerPath: packageRoot() + "/api/portfolioDrafts/funding/getFundingStep.ts",
     });
 
+    // The API spec, which just so happens to be a valid CloudFormation snippet (with some actual CloudFormation
+    // in it) gets uploaded to S3
+    const apiAsset = new s3asset.Asset(this, "ApiSpecAsset", {
+      path: "./atat_provisioning_wizard_api.yaml",
+    });
+
+    // And now we include that snippet as an actual part of the template using the AWS::Include Transform. This
+    // is where the two previous pieces come together. Now, all the Fn::Sub and other functions will be resolved
+    // which means that we can reference the ARNs of the various functions.
+    const data = cdk.Fn.transform("AWS::Include", { Location: apiAsset.s3ObjectUrl });
```

Comment:
Can we get a better name for this variable? :)

Reply:
This is the data loaded into the api spec yaml, living on s3. We could name it `apiData` , `functionData`, `resolvedLambdaArnData`, `lambdaData`, `lambdaFnNames`, `functionalDataForS3`?

Reply:
`apiSpecAsTemplateInclude`